pkg/autodetect/main_test.go

```diff
@@ -496,34 +498,96 @@ func TestCleanDeployments(t *testing.T) {
 }
 
 func TestRequireUpdates(t *testing.T) {
+	jaeger1 := v1.NewJaeger(types.NamespacedName{
+		Name:      "Jaeger1",
+		Namespace: "TestNS",
+	})
+
+	jaeger2 := v1.NewJaeger(types.NamespacedName{
+		Name:      "Jaeger2",
+		Namespace: "TestNS",
+	})
+
 	s := scheme.Scheme
+	s.AddKnownTypes(v1.SchemeGroupVersion, &v1.Jaeger{})
 	s.AddKnownTypes(v1.SchemeGroupVersion, &v1.JaegerList{})
 
 	dcl := &fakeDiscoveryClient{}
 	cl := customFakeClient()
 	b := WithClients(cl, dcl, cl)
 
-	deps := &appsv1.DeploymentList{
-		Items: []appsv1.Deployment{{
-			Spec: appsv1.DeploymentSpec{
-				Template: corev1.PodTemplateSpec{
-					Spec: corev1.PodSpec{
-						Containers: []corev1.Container{
-							{
-								Name: "my-business-container",
-							},
-							{
-								Name: "jaeger-agent", // manually added sidecar
-							},
+	dep := &appsv1.Deployment{
+		Spec: appsv1.DeploymentSpec{
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name: "my-business-container",
+						},
+					},
+				},
+			},
+		},
+	}
+	dep.Name = "my-business-container"
+
+	dep2 := &appsv1.Deployment{
+		Spec: appsv1.DeploymentSpec{
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name: "my-business-container-2",
+						},
+						{
+							Name: "jaeger-agent", // manually added sidecar
 						},
 					},
 				},
 			},
-		}},
+		},
 	}
+	dep2.Name = "my-business-container-2"
+
+	err := cl.Create(context.TODO(), jaeger1)
+	require.NoError(t, err)
+	err = cl.Create(context.TODO(), jaeger2)
+	require.NoError(t, err)
+
+	dep.Annotations = map[string]string{inject.Annotation: jaeger1.Name}
+	err = cl.Create(context.TODO(), dep)
+	require.NoError(t, err)
+
+	err = cl.Create(context.TODO(), dep2)
+
+	require.NoError(t, err)
+	err = b.detectDeploymentUpdates()
+	require.NoError(t, err)
+	persisted := &appsv1.Deployment{}
+	err = cl.Get(context.Background(), types.NamespacedName{
+		Namespace: dep.Namespace,
+		Name:      dep.Name,
+	}, persisted)
+	assert.Equal(t, 2, len(persisted.Spec.Template.Spec.Containers))
+	err = cl.Get(context.Background(), types.NamespacedName{
+		Namespace: dep2.Namespace,
+		Name:      dep2.Name,
+	}, persisted)
+	assert.NoError(t, err)
```

Comment:
From what I could understand, `dep2` will get `jaeger1` injected at this point. Please, add an assertion to make sure this is what's happening.